editor/src/core/performance/performance-regression-tests.spec.tsx

```diff
@@ -64,7 +64,7 @@ describe('React Render Count Tests -', () => {
 
     const renderCountAfter = renderResult.getNumberOfRenders()
     // if this breaks, GREAT NEWS but update the test please :)
-    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`493`)
+    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`396`)
```

Comment:
These render count changes are highly suspicious. Do you know what has caused them?

Reply:
I am also suspicious of them and have no idea why they have changed. Some went up and some went down by the same kind of amounts, which makes me _think_ it's not accounting the numbers correctly.

Reply:
It appears to relate to the use of `keepDeepReferenceIfPossible` in `applyStatePatches`, I'm changing that to only be called if there are patches to apply. As it's needed to avoid infinite React re-renders when there's changes.

Reply:
My concern is that something in this change has resulted in an actual interaction relying on the experimental code that is behind a feature flag

Reply:
Fixed in b0543d9ad8abde6cbf924fa86e7453e777928a7d.